testing/servers_test.go

```diff
@@ -0,0 +1,152 @@
+package testing
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/onsi/gomega/ghttp"
+	v1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+
+	. "github.com/onsi/ginkgo/v2/dsl/core" // nolint
+	. "github.com/onsi/gomega"             // nolint
+)
+
+var _ = Describe("RespondWithOCMObjectMarshal", func() {
+	var server *ghttp.Server
+
+	BeforeEach(func() {
+		server = MakeTCPServer()
+	})
+
+	AfterEach(func() {
+		server.Close()
+	})
+
+	It("should respond with JSON encoded object using provided marshal function", func() {
+		// Create a mock cluster object (we'll use a simple struct for testing)
+		cluster, err := v1.NewCluster().
+			ID("test-cluster").
+			Name("Test Cluster").
+			Build()
+		Expect(err).ToNot(HaveOccurred())
+
+		// Configure the server to respond with the cluster
+		server.AppendHandlers(
+			RespondWithOCMObjectMarshal(200, cluster, v1.MarshalCluster),
+		)
+
+		// Make a request
+		resp, err := http.Get(server.URL())
+		Expect(err).ToNot(HaveOccurred())
+		defer resp.Body.Close()
```

Comment:
Need to use ginko's `After` as any assertion failure will cause skipping this.

Reply:
Did you mean DeferCleanup? Changed to that just in case, using after would be more cumbersome due to setting it up as external variables and relying on closing later

Reply:
Yes, my apologies, I meant `DeferCleanup`.